packages/fab/index.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+export default class Fab extends React.Component {
+
+  state = {
+    classList: new Set(),
+  };
+
+  get classes() {
+    const {classList} = this.state;
+    const {
+      mini,
+      className,
+    } = this.props;
+
+    return classnames('mdc-fab', Array.from(classList), className, {
+      'mdc-fab--mini': mini,
+    });
+  }
+
+  render() {
+    this.children = React.Children.map(this.props.children, (item) => {
+      const className = `${item.props.className} mdc-fab__icon`;
```

Comment:
I think pulling this method out into a class method would make the logic clearer. Something like `this.addClassesToChild` or something like that.